timely/src/dataflow/channels/mod.rs

```diff
@@ -51,7 +52,6 @@ impl<T, C: Container> Message<T, C> {
 
         if let Some(message) = bundle {
             *buffer = message.data;
-            buffer.clear();
```

Comment:
This line prompted me to think I don't actually understand what is going on. It seems like a pretty substantial change to some contracts, and it's worth going through the reason for it.

Reply:
Right. Elsewhere (e.g. `columnar`) you just have a `Clear` trait and require it when you want to be able to re-use a container.